src/Microsoft.AspNetCore.Authentication.Facebook/FacebookExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authentication.Facebook;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FacebookAuthenticationOptionsExtensions
+    {
+        /// <summary>
+        /// Adds facebook authentication with options bound against the "Facebook" section 
+        /// from the IConfiguration in the service container.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services)
+            => services.AddFacebookAuthentication(FacebookDefaults.AuthenticationScheme, _ => { });
+
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services, Action<FacebookOptions> configureOptions) 
+            => services.AddFacebookAuthentication(FacebookDefaults.AuthenticationScheme, configureOptions);
+
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services, string authenticationScheme, Action<FacebookOptions> configureOptions)
+        {
+            services.AddSingleton<IConfigureOptions<FacebookOptions>, FacebookConfigureOptions>();
```

Comment:
Yeah that's where I started as well with (AddXyz() and AddXyz(o => { }) being disjoint, but given that its not possible to actually do things like bind events in the config, the delegate is going to be required most of the time, so it would be a pretty big pit of failure if they were disjoint...

I agree that might be a potential problem with no way to turn off the binding.